src/WindowsAzureMessaging/WindowsAzureMessaging/Models/MSNotificationHubMessage.m

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSNotificationHubMessage.h"
+
+@implementation MSNotificationHubMessage
+
+@synthesize title;
+@synthesize message;
+@synthesize badge;
+@synthesize additionalData;
+
+- (instancetype)init {
+    if(self = [super init])
+    {
+        additionalData = [NSMutableDictionary new];
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithNotification:(NSDictionary *)notification {
+    if (self = [super init]) {
+        additionalData = [NSMutableDictionary new];
+        
+        for (id key in notification) {
```

Comment:
See comment about not abstracting. If we want to provide some helpers for accessing things like `title` and `body`, then we can provide them as accessor methods. But we should just expose the raw notification content to the user.